viewing_party/party.py

```diff
@@ -0,0 +1,154 @@
+from collections import Counter
+
+
+def create_movie(title, genre, rating):
+    if title is None:
+        return None
+    if genre is None:
+        return None
+    if rating is None:
+        return None
+
+    movie = {}
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+
+    return movie
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            add_to_watched(user_data, movie)
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+    ratings_list = []
+
+    if len(user_data["watched"]) == 0:
+        return 0.0
+
+    for movie in user_data["watched"]:
+        ratings_list.append(movie["rating"])
+
+    average_rating = sum(ratings_list) / len(ratings_list)
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+    genre_list = []
+
+    if user_data["watched"] == []:
+        return None
+
+    for movie in user_data["watched"]:
+        genre_list.append(movie["genre"])
+
+    occurence_count = Counter(genre_list)
+    return occurence_count.most_common(1)[0][0]
```

Comment:
Python does have the Counter class to help with situations like this. We didn't discuss this during the review, but would you feel confident explaining how this works to someone else? What is `most_common(1)[0][0]` actually doing? It would be great if you had a comment with details about the research you did to arrive at this solution and your understanding of how it works!

It can be interesting to explore library types in python to save work for ourselves, but when we're getting started, generally the more we build the better, as it gives us additional practice.